x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/expression/predicate/operator/arithmetic/Arithmetics.java

```diff
@@ -5,27 +5,33 @@
  */
 package org.elasticsearch.xpack.ql.expression.predicate.operator.arithmetic;
 
-import java.time.Duration;
-import java.time.OffsetTime;
-import java.time.Period;
-import java.time.temporal.Temporal;
+import org.elasticsearch.xpack.ql.QlIllegalArgumentException;
+
+import java.util.function.BiFunction;
 
 /**
  * Arithmetic operation using the type widening rules of the JLS 5.6.2 namely
  * widen to double or float or long or int in this order.
  */
 public final class Arithmetics {
 
-    public static final long DAY_IN_MILLIS = 60 * 60 * 24 * 1000L;
-
     private Arithmetics() {}
 
-    private enum IntervalOperation {
-        ADD,
-        SUB
+    public interface NumericArithmetic extends BiFunction<Number, Number, Number> {
+        default Object wrap(Object l, Object r) {
+            if (!(l instanceof Number)) {
```

Comment:
I've moved the code as it was without doing any style touches. I'd rather do that separately (similar to organize imports) to avoid the extra noise.
I'm sure there are multiple places where the `!=` style is being used.